### ISSUE TITLE
Support for Haskel

### ISSUE THREAD

**Issue by stefanbuck** [OctoLinker/browser-extension#44]
``` haskell
import Mod
import qualified Mod
import Mod
import Mod ()
import Mod (x,y)
import qualified Mod
import qualified Mod (x,y)
import Mod hiding (x,y)
import qualified Mod hiding (x,y)
import Mod as Foo
import Mod as Foo (x,y)
import qualified Mod as Foo
import qualified Mod as Foo (x,y)
-- A lot of other combinations excluded. See the link.
-- There might also be a formal grammar somewhere.
```

https://wiki.haskell.org/Import

**Comment by sindresorhus** [comment]
// @passy

**Comment by ousennin** [comment]
I'm thinking on writing a haskell plugin. It's not hard to implement, but for me, as I'm not an experienced developer, it's hard to implement it in a way that would fit the overall architecture and style. There are some nuances with haskell. 
The path for imported libraries is whether current directory or it is specified in a file *.cabal in the main project's directory. So, we have to get the cabal file contents, if it exists, and use this regex to extract the library path: 
```/(?:(?:\n|^)library\s)(?:[\s\S]*)(?:hs-source-dirs:(?:\s))([\w\\]+)\s/gi```
For extracting imports we can use this regex:
```/(?:(?:^|\s)import\s(?:qualified\s)?)([\w.]+)/g```
We'll have to add '.hs' to the end of extracted string and replace dots for slashes to build a path.
So, this is a rough outline of how it should work. Even I fail to write my own implementation, I hope this info will be useful for other people.

**Comment by stefanbuck** [comment]
Thanks for the very detailed description. Fetching additional information from another file is not possible at the moment, but I have some ideas in mind. I think I'll pick this up soon.

**Comment by josephfrazier** [comment]
I went ahead and closed this due to #308, but we could create a new issue for reading `.cabal` files, or re-open this one, if you'd like.
